src/light_data/mod.rs

```diff
@@ -103,6 +103,10 @@ impl LightData {
     pub fn to_trimmed_le_bytes(x: usize) -> Vec<u8> {
         x.to_le_bytes()[..Self::byte_count(x) as usize].to_vec()
     }
+    /// Returns the minimum number of bytes required to represent the integer.
+    pub fn trimmed_le_bytes_len(xs: &[u8]) -> usize {
+        xs.iter().rev().fold(0, |acc, &x| (acc * 256) + x as usize)
+    }
```

Comment:
The comment doesn't match the argument (it's not an integer). It's also not returning a number of bytes, and it's unclear what the name of the function is supposed to represent.

Shouldn't we just call [`from_le_bytes`](https://doc.rust-lang.org/std/primitive.usize.html#method.from_le_bytes), which would also make the error-handling clear? This may blow up in cases where the integer is > usize, and on platforms where usize != u64.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ce46f49c20da19488389a5df317239e9